docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+tpm2_create -C 0x81010001 -u user_seal_obj_user_nv_auth.pub \
+-r user_seal_obj_user_nv_auth.priv -a "fixedtpm|fixedparent|adminwithpolicy" \
+-L policy.secret.user.nv -i-
+```
+
+* ***Unseal the wrapping secret from USER sealing object with NV pin ref DO NOT
+UNSEAL TO STDOUT***
+```
+tpm2_load -C 0x81010001 -u user_seal_obj_user_nv_auth.pub \
+-r user_seal_obj_user_nv_auth.priv -c user_seal_obj_user_nv_auth.ctx
+tpm2_startauthsession -S session.ctx --policy-session
+tpm2_policysecret -S session.ctx -c $USER_NV_INDEX \
+-L policy.secret.user.nv usernvpin
+tpm2_unseal -c user_seal_obj_user_nv_auth.ctx -p session:session.ctx
+tpm2_flushcontext session.ctx
+rm session.ctx
+tpm2_flushcontext -t
+rm user_seal_obj_user_nv_auth.ctx
+```
+
+* ***Create a sample user object with USER NV ref auth***
+```
+tpm2_create -C 0x81010001 -u user_object_user_nv_auth.pub \
+-r user_object_user_nv_auth.priv \
+-a "fixedtpm|fixedparent|sensitivedataorigin|adminwithpolicy|sign|decrypt" \
+-L policy.secret.user.nv -G aes
+```
+
+* ***Load the sample user object and use it in an operation***
+```
+#load the user object
+tpm2_load -C 0x81010001 -c user_object_user_nv_auth.ctx \
+-u user_object_user_nv_auth.pub -r user_object_user_nv_auth.priv
+
+#satisfy the policy of referred auth of the USER pin ref object
+tpm2_startauthsession -S session.ctx --policy-session
+tpm2_policysecret -S session.ctx -c $USER_NV_INDEX \
+-L policy.secret.user.nv usernvpin
+
+#perform an operation with the user object
+echo "plaintext" > test.txt
+tpm2_encryptdecrypt -c user_object_user_nv_auth.ctx test.txt -o test.enc \
+-p session:session.ctx
+
+#close auth session and clean up
+tpm2_flushcontext session.ctx 
+tpm2_flushcontext -t
+rm user_object_user_nv_auth.ctx session.ctx 
+```
+
+* ***Change USER pin with SO pin***
+```
+tpm2_startauthsession -S session.ctx --policy-session
+tpm2_policysecret -S session.ctx -c $SO_NV_INDEX defaultsonvpin
+tpm2_policycommandcode -S session.ctx TPM2_CC_NV_ChangeAuth
+tpm2_policyor -S session.ctx \
+-l sha256:policy.pass_AND_ccnvobjch,policy.secret.so.nv_AND_ccnvobjch
+tpm2_changeauth -c $USER_NV_INDEX -p session:session.ctx usernvpin_so
+tpm2_flushcontext session.ctx
+rm session.ctx
+```
+
+* ***USER PIN change with USER pin***
+```
+tpm2_startauthsession -S session.ctx --policy-session
+tpm2_policypassword -S session.ctx
+tpm2_policycommandcode -S session.ctx TPM2_CC_NV_ChangeAuth
+tpm2_policyor -S session.ctx \
+-l sha256:policy.pass_AND_ccnvobjch,policy.secret.so.nv_AND_ccnvobjch
+tpm2_changeauth -c $USER_NV_INDEX \
+-p session:session.ctx+usernvpin_so usernv_pin_user
+tpm2_flushcontext session.ctx
+rm session.ctx
+```
+
+* ***SO PIN change with SO pin***
+```
+tpm2_startauthsession -S session.ctx --policy-session
+tpm2_policypassword -S session.ctx
+tpm2_policycommandcode -S session.ctx TPM2_CC_NV_ChangeAuth
+tpm2_changeauth -c $SO_NV_INDEX -p session:session.ctx+defaultsonvpin sonvpin
+tpm2_flushcontext session.ctx
+rm session.ctx
+```
```

Comment:
The fact you took time to script it is wonderful. It allows others to now tangibly play with the thought process behind the document and either validate the reasoning or poke holes. +2